registration/registration.js

```diff
@@ -1,55 +1,90 @@
+                case `sBirthday`:
+                    let year = new Date().getFullYear() - new Date(authValue[field].replace(/\./g, `:`)).getFullYear();
+                    if (!(10 < year && year < 80)) {
+                        return `Age must be between 10 and 80 years`;
+                    }
+                    break;
+                case `sEmail`:
+                    if (!authValue[field] || !/[0-9a-zа-я_A-ZА-Я]+@[0-9a-zа-я_A-ZА-Я^.]+\.[a-zа-яА-ЯA-Z]{2,4}/i.test(authValue[field])) {
+                        return `incorrect Email address!`;
+                    }
+                    const uniqueEmail = authData.find(x => x.sEmail === authValue[field]);
+                    if (uniqueEmail) {
+                        return `email already occupied`;
+                    }
+                    break;
+                case `sPassword`:
+                    if (!authValue[field] || !/^([а-яА-ЯA-Za-z0-9]{6,})+$/.test(authValue[field])) {
+                        return `Invalid Password!`;
+                    }
+                    break;
+                case `sConfirmPassword`:
+                    if (!authValue[field] || authValue.sPassword !== authValue[field]) {
+                        return `Passwords do not match`;
+                    }
+                    break;
+            }
+        }).filter(Boolean);
 
-    for (let x in values) {
-        switch (x) {
-            case `SignupName`:
-            case `SignupSurname`:
-                if (!/^([а-яА-ЯA-Za-z]{2,20})+$/.test(values[x])) {
-                    return warning(`Некорректный ввод!`);
-                }
-                break;
-            case `SignupBirthday`:
-                let year = new Date().getFullYear() - new Date(values[x].replace(/\./g, `:`)).getFullYear();
-                if (!(10 < year && year < 80)) {
-                    return warning(`не младше 10 и не старше 80 лет`);
-                }
-                break;
-            case `SignupEmail`:
-                if (!/[0-9a-zа-я_A-ZА-Я]+@[0-9a-zа-я_A-ZА-Я^.]+\.[a-zа-яА-ЯA-Z]{2,4}/i.test(values[x])) {
-                    return warning(`Ошибка ввода Email-адреса!`);
-                }
-                break;
-            case `SignupPassword`:
-                if (!/^([а-яА-ЯA-Za-z0-9]{6,})+$/.test(values[x])) {
-                    return warning(`Invalid Password!`);
-                }
-                break;
-            case `SignupConfirmPassword`:
-                if (values.SignupPassword !== values[x]) {
-                    return warning(`Пароли не совпадают`);
-                }
-                break;
-        }
+    if (warnings.length) {
+        warning(warnings);
+        return false;
+    } else {
+        authData.push(authValue);
+        localStorage.setItem(`authData`, JSON.stringify(authData));
+        return true;
     }
-    function warning(x) {
-        return !(document.getElementsByClassName(`warning`)[0].textContent = x);
+
+    function warning(messages) {
```

Comment:
I saw this the same function under. Exclude it from this and that file, create `utils.js` and paste it. This is global function that should be used everywhere.